tests/shared/pages/InboxPage.ts

```diff
@@ -0,0 +1,68 @@
+import { type Locator, type Page, expect } from '@playwright/test'
+
+import { BASE_URL } from '../constants'
+
+export class ModalPage {
+  private readonly baseURL = BASE_URL
+
+  private readonly connectButton: Locator
+
+  constructor(public readonly page: Page) {
+    this.connectButton = this.page.getByRole('button', { name: 'Connect Wallet' })
+  }
+
+  async load() {
+    await this.page.goto(this.baseURL)
+  }
+
+  async copyConnectUriToClipboard() {
+    await this.page.goto(this.baseURL)
+    await this.connectButton.click()
+    await this.page.getByTestId('wallet-selector-walletconnect').click()
+    await this.page.waitForTimeout(2000)
+    await this.page.getByTestId('copy-wc2-uri').click()
+  }
+
+  async disconnect() {
+    await this.page.getByTestId('account-button').click()
+    await this.page.getByTestId('disconnect-button').click()
+  }
+
+  async promptSiwe() {
+    await this.page.getByRole('button', { name: 'Sign in with wallet' }).click()
+  }
+
+  async rejectNotifications() {
+    // Allow for the modal to pop up
+    await this.page.waitForTimeout(4000)
+    const isVisible = (await this.page.locator('.NotificationPwaModal__close-button').count()) > 0
+    if (!isVisible) return
+    await this.page.locator('.NotificationPwaModal__close-button').first().click()
+  }
+
+  async subscribe(nth: number) {
+    await this.page.locator('.AppCard__body > .AppCard__body__subscribe').nth(nth).click()
+    await this.page.getByText('Subscribed to', { exact: false }).isVisible()
+  }
+
+  async unsubscribe(nth: number) {
```

Comment:
This API will flake in future revisions because number is non-deterministic. Better to select by name.